### DISCUSSION TITLE
How to get org.osgi.service.component resolved?

### DISCUSSION THREAD

**Discussion by mickaelistria** [eclipse-m2e/m2e-core#1403] [Q&A] [answered]
I'm trying to work on org.eclipse.m2e.jdt bundle (and only this one). I've imported it as usual, and project is imported according to the .project. However, I get errors that types from `org.osgi.service.component` cannot be resolved. Indeed, there is no bundle contributing org.osgi.service.component in the "Plugin Dependencies". Is there any special trick that needs to be performed to make it run? If those are only devtime dependencies, shouldn't they still be added to build.properties/additional.bundles?

**Comment by laeubi** [comment]
This is provided by PDE (I think from 2022-06 onwards), so you should make sure to use a recent eclipse build.

**Reply by mickaelistria** [reply]
I have a recent Eclipse build, latest I-Build of today to be precise. But still no magic happening. Is there a flag or something I need to check?

**Reply by laeubi** [reply, marked as answer]
It must be a PDE project and the target-state must contain `org.osgi.service.component.annotations` bundle, that should be enough.
